lib/ui/platform_dispatcher.dart

```diff
@@ -2605,3 +2621,37 @@ class SemanticsActionEvent {
     );
   }
 }
+
+/// Callback to subscribe to [ViewFocusChange] notifications.
+typedef ViewFocusChangeCallback = void Function(ViewFocusChange);
+
+/// An event to communicate view focus changes between the framework and engine.
+///
+/// If the event comes from the framework it means there was a flutter focus change.
+/// If the event comes from the engne it means there was a platform focus change.
+final class ViewFocusChange {
+  /// Creates a [ViewFocusChange].
+  const ViewFocusChange({
+    required this.viewId,
+    required this.viewFocusState,
+  });
+
+  /// The id of the [FlutterView] that experienced the focus change.
+  final int viewId;
+
+  /// The current [FocusState] of the view.
+  final ViewFocusState viewFocusState;
+}
+
+/// Represents a given flutter view focus state.
+///
+/// If focus was lost, use [ViewFocusState.unfocused].
+/// If focus was gained without direction, use [ViewFocusState.focused].
+/// If focus was gained in forward direction, use [ViewFocusState.forwardFocused].
+/// If focus was gained in backwards direction, use [ViewFocusState.backwardsFocused].
+enum ViewFocusState {
+  unfocused,
```

Comment:
I forget: for unfocused the direction is irrelevant? Don't we need to know whether the element before or after the flutter view should be focused next?

Reply:
Not sure if it is the case in other platforms. In web, at least, unfocusing a given DOM node (programmatically) means the focus will go back to the `<body />` of the document. 

cc @yjbanov to keep me honest.

Reply:
PTAL